test/powershell/Modules/Microsoft.PowerShell.Security/CmsMessage.Tests.ps1

```diff
@@ -271,16 +271,28 @@ Describe "CmsMessage cmdlets thorough tests" -Tags "Feature" {
         $errors[0].FullyQualifiedErrorId | Should Be "NoCertificateFound"
     }
 
-    It "Verify error when encrypting to wrong cert" {
-        $errors = $null
-        $goodCerts = @(Get-ChildItem Cert:\currentuser\My -DocumentEncryptionCert | ForEach-Object Thumbprint)
-        $badCert = Get-ChildItem Cert:\currentuser\My | ? { $_.Thumbprint -notin $goodCerts } | Select -First 1
+    Context "Verify error when encrypting to wrong cert" {
+        BeforeAll {
+            if ($IsWindows)
+            {
+                $goodCerts = @(Get-ChildItem Cert:\currentuser\My -DocumentEncryptionCert | ForEach-Object Thumbprint)
+                $badCert = Get-ChildItem Cert:\currentuser\My | ? { $_.Thumbprint -notin $goodCerts } | Select -First 1
+                $skipTest = $badCert -eq $null ## No proper Cert found.
```

Comment:
Does this indicate a failure during setup? I'm a little worried that this line will silently hide a setup failure that won't be noticed because the test will fall into the "skipped" bucket.